terraform/environment/kubernetes.cluster.vars.tf

```diff
@@ -0,0 +1,29 @@
+# FUTUREWORK: replace 'any' by implementing https://www.terraform.io/docs/language/functions/defaults.html
+#
+variable "k8s_cluster" {
+  description = "represents Kubernetes cluster"
+  # type = object({
+  #   cloud = string
+  #   load_balancer = optional(bool)
+  #   load_balancer_ports = optional(list(
+  #     object({
+  #       name = string
+  #       protocol = string
+  #       listen = number
+  #       destination = number
+  #     })
+  #   ))
+  #   machine_groups = list(object({
+  #     group_name = string
+  #     machine_count = optional(number)
+  #     machine_type = string
+  #     component_classes = list(string)
+  #     volume = optional(object({
+  #       size = number
+  #       format = optional(string)
+  #     }))
+  #   }))
+  # })
+  type = any
+  default = {}
+}
```

Comment:
Why do we want to encode all kinds of k8s_clusters in one type of specification? I have a feeling that this will bite us in future as we might get into specifying AWS specific things somehow. Maybe it is better to just have this variable be called "hetzner_k8s_cluster". When we start providing an option to support AWS, we can just create another variable called "aws_k8s_cluster"?

Reply:
We should also ensure that we don't create even number of etcds as it provide the same fault tolerance as creating 1 less etcd.